src/leaflet.filelayer.js

```diff
@@ -63,46 +63,117 @@
             var ext;
             var reader;
             var parser;
+
+            // Check file is defined
+            if (this._isParameterMissing(file, 'file')) {
+                return;
+            }
+
             // Check file size
-            var fileSize = (file.size / 1024).toFixed(4);
-            if (fileSize > this.options.fileSizeLimit) {
-                this.fire('data:error', {
-                    error: new Error(
-                        'File size exceeds limit (' +
-                        fileSize + ' > ' +
-                        this.options.fileSizeLimit + 'kb)'
-                    )
-                });
+            if (!this._isFileSizeOk(file.size)) {
                 return;
             }
 
-            // Check file extension
-            ext = file.name.split('.').pop();
-            parser = this._parsers[ext];
+            // Get parser for this data type
+            parser = this._getParser(file.name);
             if (!parser) {
-                this.fire('data:error', {
-                    error: new Error('Unsupported file type ' + file.type + '(' + ext + ')')
-                });
                 return;
             }
+
             // Read selected file using HTML5 File API
             reader = new FileReader();
             reader.onload = L.Util.bind(function (e) {
                 var layer;
                 try {
-                    this.fire('data:loading', { filename: file.name, format: ext });
-                    layer = parser.call(this, e.target.result, ext);
-                    this.fire('data:loaded', { layer: layer, filename: file.name, format: ext });
+                    this.fire('data:loading', { filename: file.name, format: parser.ext });
+                    layer = parser.processor.call(this, e.target.result, parser.ext);
+                    this.fire('data:loaded', { layer: layer, filename: file.name, format: parser.ext });
```

Comment:
This line exceeds the 100 char, please pass the options object on multiple lines